### PR TITLE
[SW-2087] Deprecate MetricsSupport and make it possible to obtain training metrics on H2OMojoModel

### PR DESCRIPTION
**Pull request by jakubhava**
JIRA for removal: https://0xdata.atlassian.net/browse/SW-2088

### CODE REVIEW

File: py/src/ai/h2o/sparkling/ml/params/H2OTypeConverters.py [hunk 1]
```diff
@@ -300,3 +301,17 @@ def scalaMapToDictStringToArrayString(value):
...
+            raise TypeError("None is not allowed.")
+        elif isinstance(value, JavaObject):
+            it = value.toIterator()
+            map = {}
+            while it.hasNext():
+                pair = it.next()
+                map[pair._1()] = pair._2()
+            return map
+        else:
+            raise TypeError("Invalid type.")
```
Comment (mn-mikke) [inline py/src/ai/h2o/sparkling/ml/params/H2OTypeConverters.py]: nit: missing new line

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala [hunk 2]
```diff
@@ -105,25 +137,90 @@ trait H2OMOJOModelUtils {
...
+      json.remove("domains")
+      json.remove("help")
       val gson = new GsonBuilder().setPrettyPrinting().create
-      val prettyJson = gson.toJson(modelOutputJson)
+      val prettyJson = gson.toJson(json)
       prettyJson
     }
   }
 
+  protected def extractMetrics(json: JsonObject, metricType: String): Map[String, Double] = {
```
Comment (mn-mikke) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala]: Has anything changed in the methods that were moved from `H2OModel`?

Comment (jakubhava) [reply scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala]: Nope, complete copy

**Comment by jakubhava** [comment]
> Shouldn't the changes be also reflected in the documentation?

Good catch, will update the Mojo doc

File: py/src/ai/h2o/sparkling/ml/params/H2OTypeConverters.py (hunk 1 above)
Comment (jakubhava) [reply py/src/ai/h2o/sparkling/ml/params/H2OTypeConverters.py]: 👍

**Pull request merged**